flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/util/AWSUtil.java

```diff
@@ -50,8 +50,14 @@ public static AmazonKinesisClient createKinesisClient(Properties configProps) {
 		awsClientConfig.setUserAgent("Apache Flink " + EnvironmentInformation.getVersion() +
 			" (" + EnvironmentInformation.getRevisionInformation().commitId + ") Kinesis Connector");
 
-		AmazonKinesisClient client =
-			new AmazonKinesisClient(AWSUtil.getCredentialsProvider(configProps).getCredentials(), awsClientConfig);
+		AWSCredentialsProvider credentialsProvider = AWSUtil.getCredentialsProvider(configProps);
+
+		if (credentialsProvider != null) {
+			AmazonKinesisClient client =
+				new AmazonKinesisClient(credentialsProvider.getCredentials(), awsClientConfig);
+		} else {
+			AmazonKinesisClient client = new AmazonKinesisClient(awsClientConfig);
+		}
```

Comment:
No problem at all :) Thank you for the quick fix.
Can you also revert back the declaration of `credentialsProvider`? I don't think it is required to declare this variable outside the scope.